apps/ledger-live-mobile/src/screens/CustomImage/Step1Crop.tsx

```diff
@@ -0,0 +1,188 @@
+ * Then on confirmation it navigates to the preview step with the cropped image
+ * file URI as a param.
+ */
+const Step1Cropping: React.FC<{}> = () => {
+  const cropperRef = useRef<CropView>(null);
+  const [imageToCrop, setImageToCrop] = useState<
+    (ImageFileUri & Partial<ImageDimensions>) | null
+  >(null);
+  const [rotated, setRotated] = useState(false);
+
+  const { t } = useTranslation();
+
+  const navigation = useNavigation();
+  const { params }: { params: RouteParams } = useRoute();
+
+  const handleError = useCallback(
+    (error: Error) => {
+      console.error(error);
+      navigation.navigate(ScreenName.CustomImageErrorScreen, { error });
+    },
+    [navigation],
+  );
+
+  /** LOAD SOURCE IMAGE FROM PARAMS */
+  useEffect(() => {
+    loadImageToFileWithDimensions(params)
+      .then(res => {
+        setImageToCrop(res);
+      })
+      .catch(handleError);
+  }, [params, setImageToCrop, handleError]);
+
+  /** CROP IMAGE HANDLING */
+  const handleCropResult: ImageCropperProps["onResult"] = useCallback(
+    (res: CropResult) => {
+      navigation.navigate(ScreenName.CustomImageStep2Preview, res);
+    },
+    [navigation],
+  );
+
+  const handlePressNext = useCallback(() => {
+    cropperRef?.current?.saveImage(undefined, 100);
+  }, [cropperRef]);
+
+  const handlePressRotateLeft = useCallback(() => {
+    if (cropperRef?.current) {
+      cropperRef.current.rotateImage(false);
+      setRotated(!rotated);
+    }
+  }, [cropperRef, rotated, setRotated]);
+
+  const [
+    containerDimensions,
+    setContainerDimensions,
+  ] = useState<ImageDimensions | null>(null);
+  const onContainerLayout = useCallback(({ nativeEvent: { layout } }) => {
+    setContainerDimensions({ height: layout.height, width: layout.width });
+  }, []);
+
+  const sourceDimensions = useMemo(
+    () =>
+      fitImageContain(
+        {
+          height:
+            (Platform.OS === "android" && rotated
+              ? imageToCrop?.width
+              : imageToCrop?.height) ?? 200,
+          width:
+            (Platform.OS === "android" && rotated
+              ? imageToCrop?.height
+              : imageToCrop?.width) ?? 200,
+        },
+        boxToFitDimensions,
+      ),
+    [imageToCrop?.height, imageToCrop?.width, rotated],
+  );
+
+  const sourceAspectRatio =
+    sourceDimensions.height / (sourceDimensions.width || 1);
+  const imageCropperStyle =
+    Math.max(sourceAspectRatio, 1 / (sourceAspectRatio || 1)) > 2
+      ? containerDimensions
+      : sourceDimensions;
+
+  return (
+    <Flex flex={1}>
+      <Flex
+        flex={1}
+        flexDirection="column"
+        alignItems="center"
+        justifyContent="center"
+      >
+        <Flex
+          flex={1}
+          onLayout={onContainerLayout}
+          width="100%"
+          justifyContent="center"
+          alignItems="center"
+        >
+          {(true || false) && imageToCrop ? (
```

Comment:
what is this black magic ? 🧙‍♂️

Reply:
LOL wtf I do that sometimes when debugging UI 😆 this one is really weird though

Reply:
i really hope you don't find more ridiculous stuff like this